X16 Reference - Appendix C - 65C02 Processor.md

```diff
@@ -1,155 +1,224 @@
-
 # Appendix C: The 65C02 Processor
 
-This is not meant to be a complete manual on the 65C02 processor, though is meant 
-to serve as a convenient quick reference. Much of this information comes from
-6502.org and pagetable.com. It is been placed here for convenience though further
-information can be found at those (and other) sources.
-
-## Work In Progress
+This is not meant to be a complete manual on the 65C02 processor, though is
+meant to serve as a convenient quick reference. Much of this information comes
+from 6502.org and pagetable.com. It is been placed here for convenience though
+further information can be found at those (and other) sources.
 
-TODO:
+## Overview
 
-  * Add 65C02 addressing modes for ADC AND CMP EOR LDA ORA SBC STA
-  * Add flag changes where relevant (likely missing some)
+The WDC65C02 CPU is a modern version of the MOS6502 with a few additional
+instructions and addressing modes and is capable of running at up to 14 MHz. On
+the Commander X16, it is clocked at 8 MHz.
 
-## Possible 65C816 Support Compatibilty
+## A note about 65C816 Compatibilty
 
 The Commander X16 may be upgraded at some point to use the WDC 65C816 CPU.
-The 65C816 is mostly compatible with the 65C02, except for 4 instructions 
+The 65C816 is mostly compatible with the 65C02, except for 4 instructions
 (`BBRx`, `BBSx`, `RMBx`, and `SMBx`).
 
-As a result, we recommend *not* using the BBR, BBS, RMB and SMB instructions, as these
-may become invalid in future versions of the Commander X16. 
-
-These instructions are *not* supported on the Commander X16 as of the R47 release.
-
-## Overview
-
-The WDC65C02 CPU is a modern version of the MOS6502 with a few additional instructions
-and addressing modes and is capable of running at up to 14 MHz. On the Commander X16
-it is clocked at 8 MHz.
+These instructions are *not* supported on the Commander X16 as of the R47
+release and will generate an error when used in the emulator.
```

Comment:
I think a deprecation cycle would be in order here for the emu.  These are still valid instructions on the hardware as formerly announced and delivered, and I hate to break people's working code as a suprrise.  I think the first change would be to issue a warning on the console when the instructions are used, and offer a command line switch to suppress the warnings.

Once the emulator has 65C816 support, it might be more appropriate to produce an error while the emulator is running in 65C02 mode (without the given flag).  In 65C816 mode, the instructions will of course have another meaning.

Perhaps we can change "an error" to "a warning" here?

Regardless, I think users should be given the ability to run existing code with newer emus, even deprecated code, otherwise some of them will be stubborn and refuse to upgrade entirely.

Reply:
No problem. I'll roll that back to something like:

These instructions *may* be removed from the emulator in a future release cycle. This will be communicated well in advance, after a decision is made about any possible 65C816 transition.